### PR TITLE
CASMCMS-8644: Move from v6.4.0 to v6.6.0 of openapi-generator

### PR DESCRIPTION
**Pull request by mharding-hpe**
## Summary and Scope

We are currently using v6.4.0 of the generator. It looks like v6.6.0 is the latest release. I think we should at least update to v6.5.0, because the release notes for that indicate that it improves support for `anyOf` and `allOf` OpenAPI keywords, and we're now making more use of those.

This PR just bumps the generator version used for generation in the Dockerfile.

## Testing

I ran the cmsdev regression tests and they passed.

## Risks and Mitigations

Low risk -- just a small version bump on the API generator.

## Pull Request Checklist

- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [X] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by mharding-hpe** [comment]
This is ready to merge once it receives approvals.

**Review by rbak-hpe — APPROVED** [review]

**Pull request merged**
